src/mongocxx/test/versioned_api.cpp

```diff
@@ -0,0 +1,200 @@
+    const auto client_opts =
+        options::client{}
+            .server_api_opts(server_api_opts); // Set the version and options
+    // Create a new client with the options
+    mongocxx::client client{client_uri, client_opts};
+    // End Versioned API Example 2
+}
+
+TEST_CASE("Versioned API, non-strict, for commands/features outside versioned API") {
+    instance::current();
+    if (!is_server_v5_or_newer()) {
+        return;
+    }
+    // Start Versioned API Example 3
+    using namespace mongocxx;
+    uri client_uri{"mongodb://localhost"};
+    // Create an option set for API v1
+    const auto server_api_opts =
+        options::server_api{options::server_api::version_from_string("1")}
+            .strict(false);  // Explicitly disable strict mode for the server API
+    // Store it in the set of client options
+    const auto client_opts =
+        options::client{}
+            .server_api_opts(server_api_opts); // Set the version and options
+    // Create a new client with the options
+    mongocxx::client client{client_uri, client_opts};
+    // End Versioned API Example 3
+}
+
+TEST_CASE("Versioned API, non-strict with deprecation errors") {
+    instance::current();
+    if (!is_server_v5_or_newer()) {
+        return;
+    }
+    // Start Versioned API Example 4
+    using namespace mongocxx;
+    uri client_uri{"mongodb://localhost"};
+    // Create an option set for API v1
+    const auto server_api_opts =
+        options::server_api{options::server_api::version_from_string("1")}
+            .deprecation_errors(true);  // Enable deprecation errors
+    // Store it in the set of client options
+    const auto client_opts =
+        options::client{}
+            .server_api_opts(server_api_opts); // Set the version and options
+    // Create a new client with the options
+    mongocxx::client client{client_uri, client_opts};
+    // End Versioned API Example 4
+}
+// clang-format on
+
+TEST_CASE("Versioned API, with insert-many for 'count' migration") {
+    instance::current();
+    if (!is_server_v5_or_newer()) {
+        return;
+    }
+
+    using namespace mongocxx;
+    mongocxx::client client{
+        uri{},
+        options::client{}.server_api_opts(
+            options::server_api{options::server_api::version::k_version_1}.strict(true))};
+    using namespace bsoncxx::builder::basic;
+
+    // Drop in case we have stale data
+    client.database("db").collection("sales").drop();
+
+    // clang-format off
+    // Start Versioned API Example 5
+    auto db = client.database("db");
+    auto sales = db.collection("sales");
+    auto make_sale =
+        [](int id, std::string item_name, int price, int quanitity, std::string time) {
+            return make_document(kvp("_id", id),
+                                 kvp("item", item_name),
+                                 kvp("price", price),
+                                 kvp("quantity", quanitity),
+                                 kvp("date", time));
```

Comment:
The `date` should be a BSON date type, not a string. It is represented as a unix timestamp. I don't think converting ISO date strings to Unix timestamps is relevant to the example. But for consistency of types between language examples, I suggest adding a helper like `iso_to_unix` that is defined outside of the example, similar to the one defined in C https://github.com/mongodb/mongo-c-driver/commit/0d6a8ba27ed5cd94a26e4db879f61b545799193f#diff-803a49bf3111998fa92319347cedc6c44fe5070ed21e30d4bcc03b87653d880aR3706. It can bypass any actual conversion, but help signify to the user that a BSON date is being stored.

Aside: https://bsonspec.org/spec.html is the canonical reference on BSON types.